### PR TITLE
[eiger] Update spack configs for eiger

### PR DESCRIPTION
**Pull request by rasolca**
Version of modules and packages has been updated to match what is available after the maintenance.

### CODE REVIEW

**Comment by lucamar** [comment]
ok to test

File: spack/eiger/compilers.yaml [hunk 1]
```diff
@@ -516,12 +154,12 @@ compilers:
     operating_system: sles15
     target: any
     modules:
-    - PrgEnv-gnu
-    - gcc/9.3.0
+    - PrgEnv-intel
+    - intel/8.0.0
```
Comment (haampie) [inline spack/eiger/compilers.yaml:158]: This doesn't exist, I guess spack doesn't detect the right thing here..

File: spack/eiger/compilers.yaml [hunk 2]
```diff
@@ -1,422 +1,60 @@
...
-    paths:
-      cc: cc
-      cxx: CC
-      f77: ftn
-      fc: ftn
-    spec: pgi@18.5.0
-    target: any
-- compiler:
+    - PrgEnv-aocc
+    - aocc/8.0.0
```
Comment (haampie) [inline spack/eiger/compilers.yaml:17]: Same, doesn't exist...

**Comment by haampie** [comment]
I've pushed some more changes. I can revert the `buildable: false` bits, but in my experience with spack using system libs by default was more often asking for problems than that it would make builds shorter.

**Comment by rasolca** [comment]
Thank you @haampie.
Regarding `buildable: false` I kept them from the daint config.

**Review by victorusu — APPROVED** [review]
`lgtm`

**Comment by victorusu** [comment]
@haampie, can we merge? what do you think?

**Comment by haampie** [comment]
Yeah, should be good to go

**Pull request merged**